### PR TITLE
IN: Pet Meetup Events

### PR DESCRIPTION
**Pull request by serafy0**
### What this PR does (required):
- merged both https://github.com/hatchways/team-naan-bread/pull/90 and https://github.com/hatchways/team-naan-bread/pull/91 and added updates to integrate them together
- updated the frontend with components and APICalls to edit, delete or attend an event
- added a route in the backend to fetch one event with populated attendees and host.
- updated the event model in both the frontend and backend to use MongoDB's [GeoJSON](https://mongoosejs.com/docs/geojson.html)
### Screenshots / Videos (required):
https://user-images.githubusercontent.com/29197126/135327617-5d8f722c-8b9f-4e0b-ba05-fc6b12fa7aba.mov

### Any information needed to test this feature (required):
- log in as a new user
-  click on events item in the navbar to add a new event or see events nearby.
- if you allow geolocation in the `/events` page, you should fetch all events with in 100km from your location

### CODE REVIEW

File: client/src/components/PetEvent/DeleteEventButton.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const deleteEvent = async () => {
+    console.log('yes yes yes');
```
Comment (ezeilo-su) [inline client/src/components/PetEvent/DeleteEventButton.tsx:28]: Remove console.log to keep your code cleaner.

File: server/models/Event.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
 const mongoose = require('mongoose');
+
+const pointSchema = new mongoose.Schema({
+  type: {
+    type: String,
+    enum: ['Point'],
+    required: true,
+  },
+  //Note that longitude comes first in a GeoJSON coordinate array, not latitude
```
Comment (ezeilo-su) [inline server/models/Event.js:9]: Please remove comment

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Since you merged this with the previous PR, it's good to rebase your branch properly so that changes in the previous PR won't show up again as new changes in this PR.

**Comment by serafy0** [comment]
> Since you merged this with the previous PR, it's good to rebase your branch properly so that changes in the previous PR won't show up again as new changes in this PR.

Thank you so much for the advice,
I'm not sure how can I remove and rebase https://github.com/hatchways/team-naan-bread/pull/92/commits/46b94f5c6bc31f1dcafbd7d3b812b88f626610d0 https://github.com/hatchways/team-naan-bread/pull/92/commits/576de9a0d7132f562f0cb614c4632d8c3308e02c. Is it safe to try to cherry-pick them with `cherry pick -m` and drop them, Then pull my changes again using rebase ?

**Comment by ezeilo-su** [comment]
> > Since you merged this with the previous PR, it's good to rebase your branch properly so that changes in the previous PR won't show up again as new changes in this PR.
> 
> Thank you so much for the advice, I'm not sure how can I remove and rebase [46b94f5](https://github.com/hatchways/team-naan-bread/commit/46b94f5c6bc31f1dcafbd7d3b812b88f626610d0) [576de9a](https://github.com/hatchways/team-naan-bread/commit/576de9a0d7132f562f0cb614c4632d8c3308e02c). Is it safe to try to cherry-pick them with `cherry pick -m` and drop them, Then pull my changes again using rebase ?

You can factor this in your future PRs

**Comment by serafy0** [comment]
When I tried to pull and rebase my resolved merge conflicts from #90  into this PR. the get client detected new merge conflicts in this branch, so I had to merge it again here

File: server/models/Event.js [hunk 3]
```diff
@@ -1,12 +1,25 @@
 const mongoose = require('mongoose');
+
+const pointSchema = new mongoose.Schema({
+  type: {
+    type: String,
+    enum: ['Point'],
+    required: true,
```
Comment (moffatethan) [inline server/models/Event.js:7]: Not sure the purpose here, enum with one item?

**Review by moffatethan — APPROVED** [review]
Really good work here - great job!

File: server/models/Event.js (hunk 3 above)
Comment (serafy0) [reply server/models/Event.js:7]: I took it from the mongoose docs, I think [GeoJSON](https://mongoosejs.com/docs/geojson.html) points needs to be in this format in order to work

**Pull request merged**